File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/ConstantEditorFragment.java
package ch.rmy.android.http_shortcuts.variables.types;

import android.widget.EditText;

import butterknife.Bind;
import ch.rmy.android.http_shortcuts.R;
import ch.rmy.android.http_shortcuts.realm.models.Variable;

public class ConstantEditorFragment extends VariableEditorFragment {

    @Bind(R.id.input_variable_value)
    EditText value;

    @Override
    protected int getLayoutResource() {
        return R.layout.variable_editor_constant;
    }

    @Override
    public void updateViews(Variable variable) {
        value.setText(variable.getValue());
    }

    @Override
    public void compileIntoVariable(Variable variable) {
        variable.setValue(value.getText().toString());
    }
}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/variables/types/ConstantEditorFragment.kt
package ch.rmy.android.http_shortcuts.variables.types

import android.view.LayoutInflater
import android.view.ViewGroup
import ch.rmy.android.http_shortcuts.data.models.Variable
import ch.rmy.android.http_shortcuts.databinding.VariableEditorConstantBinding
import ch.rmy.android.http_shortcuts.variables.VariableViewUtils.bindVariableViews

class ConstantEditorFragment : VariableEditorFragment<VariableEditorConstantBinding>() {

    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?) =
        VariableEditorConstantBinding.inflate(inflater, container, false)

    override fun setupViews() {
        bindVariableViews(binding.inputVariableValue, binding.variableButton, variablePlaceholderProvider)
            .attachTo(destroyer)
    }

    override fun updateViews(variable: Variable) {
        binding.inputVariableValue.rawString = variable.value ?: ""
    }

    override fun compileIntoVariable(variable: Variable) {
        variable.value = binding.inputVariableValue.rawString
    }
}
